### ISSUE TITLE
Feature Request: top level operator `$expr`

### ISSUE THREAD

**Issue by valerierose** [mongomock/mongomock#535]
I have an aggregate query that includes this `$match` action in the pipeline:

```
{'$match': {'$or': [{'$expr': {'$gt': ["$latest", "$thing.analysisMeta.timestamp"]}},
                            {'thing.analysisMeta': None}]}},
```

In the real mongo database, this aggregate query works. With MongoMock >= 3.15.0, I get the error:

```
pymongo.errors.OperationFailure: unknown top level operator: $expr
```

In MongoMock 3.12.0 and 3.14.0, I do not get the error. I think it might be related to the fix for this issue: https://github.com/mongomock/mongomock/issues/495

For now, I can stay on 3.14.0, but would be nice to fix this in future releases so that we are not stuck.

**Comment by pcorpet** [comment]
Hi @valerierose , thanks for reporting. It actually never worked because it was never implemented in `mongomock`. However it should not raise an `OperationFailure` but a `NotImplementedError`. I'll fix that ASAP, and then try an implementation of the feature but that might take more time.

**Comment by pcorpet** [comment]
I've fixed the type of the error in e5aeccb, so it's not a bug anymore but a feature request.
